src/pages/Pools.tsx

```diff
@@ -5,49 +5,113 @@ import {
+        poolData: d4PoolData,
+        userShareData: d4UserShareData,
+        poolRoute: "/pools/d4",
+      }
+    } else if (poolName === ALETH_POOL_NAME) {
+      return {
+        poolData: alethPoolData,
+        userShareData: alethUserShareData,
+        poolRoute: "/pools/aleth",
+      }
+    } else if (poolName === BTC_POOL_NAME) {
+      return {
+        poolData: btcPoolData,
+        userShareData: btcUserShareData,
+        poolRoute: "/pools/btc",
+      }
+    } else if (poolName === STABLECOIN_POOL_NAME) {
+      return {
+        poolData: usdPoolData,
+        userShareData: usdUserShareData,
+        poolRoute: "/pools/usd",
+      }
+    } else {
+      return {
+        poolData: veth2PoolData,
+        userShareData: veth2UserShareData,
+        poolRoute: "/pools/veth2",
+      }
+    }
+  }
   return (
     <div className={styles.poolsPage}>
       <TopMenu activeTab="pools" />
+      <ul className={styles.filters}>
+        <li
+          className={classNames(styles.filterTab, {
+            [styles.selected]: filter === "all",
+          })}
+          onClick={(): void => setFilter("all")}
+        >
+          ALL
+        </li>
+        <li
+          className={classNames(styles.filterTab, {
+            [styles.selected]: filter === PoolTypes.BTC,
+          })}
+          onClick={(): void => setFilter(PoolTypes.BTC)}
+        >
+          BTC
+        </li>
+        <li
+          className={classNames(styles.filterTab, {
+            [styles.selected]: filter === PoolTypes.USD,
+          })}
+          onClick={(): void => setFilter(PoolTypes.USD)}
+        >
+          USD
+        </li>
+        <li
+          className={classNames(styles.filterTab, {
+            [styles.selected]: filter === PoolTypes.ETH,
+          })}
+          onClick={(): void => setFilter(PoolTypes.ETH)}
+        >
+          ETH
+        </li>
+        <li
+          className={classNames(styles.outdated, styles.filterTab, {
+            [styles.selected]: filter === "outdated",
+          })}
+          onClick={(): void => setFilter("outdated")}
+        >
+          OUTDATED
+        </li>
+      </ul>
+
       <div className={styles.content}>
-        <PoolOverview
-          poolData={d4PoolData}
-          poolRoute={`/pools/d4`}
-          userShareData={d4UserShareData}
-        />
-        <PoolOverview
-          poolData={alethPoolData}
-          poolRoute={`/pools/aleth`}
-          userShareData={alethUserShareData}
-        />
-        <PoolOverview
-          poolData={btcPoolData}
-          poolRoute={`/pools/btc`}
-          userShareData={btcUserShareData}
-        />
-        <PoolOverview
-          poolData={usdPoolData}
-          poolRoute={`/pools/usd`}
-          userShareData={usdUserShareData}
-        />
-        <PoolOverview
-          poolData={veth2PoolData}
-          poolRoute={`/pools/veth2`}
-          userShareData={veth2UserShareData}
-        />
+        {Object.values(POOLS_MAP)
```

Comment:
this places the alETH and D4 pools at the very bottom, can we update it to preserve the current ordering (sorted by TVL)

Reply:
```
{
Object.values(POOLS_MAP)
.filter(...)
.map(({ name }) => getPropsForPool(name))
.sort((a,b) => {
  // 1. active pools 
  // 2. user pools
  // 3. higher TVL pools
  if (a.poolData?.migrate != null) {
    return 1
  } else if (b.poolData?.migrate != null) {
    return -1
  } else if (a.userShareData?.usdBalance) {
    if (b.userShareData?.usdBalance) {
      return a.userShareData.usdBalance.gt(b.userShareData.usdBalance) ? -1 : 1
    }
    return -1
  } else if (a.poolData?.reserve) {
    if (b.poolData?.reserve) {
      return a.poolData.reserve.gt(b.poolData.reserve) ? -1 : 1
    }
    return -1
  }
  return -1
}) 
}
.map(poolProps => <PoolOverview key={poolProps.name} {...poolProps} />) // TODO add name to poolProps
```